Dockerfile

```diff
@@ -1,40 +1,34 @@
-ARG RUBY_VERSION=3.0
-FROM ruby:${RUBY_VERSION}-buster@sha256:6986a308eab6b20c612e6e0376196fa05e79973dd6e42c588c498dee9ce54832 as dev
+FROM ruby:3.0-buster@sha256:6986a308eab6b20c612e6e0376196fa05e79973dd6e42c588c498dee9ce54832 AS base
 
 ENV TZ=US/Pacific
-
-ARG NODE_VERSION=17
-ENV NODE_VERSION=$NODE_VERSION
-
-RUN curl -sL https://deb.nodesource.com/setup_${NODE_VERSION}.x -o node_setup.sh && \
-      bash node_setup.sh 1> /dev/null
 RUN apt-get update && apt-get install -yq --no-install-recommends \
       build-essential \
-      vim \
-      git \
       diffutils \
+      git \
+      lsof \
+      vim-nox \
       xdg-user-dirs \
-      nodejs && \
-      rm -rf /var/lib/apt/lists/*
+    && rm -rf /var/lib/apt/lists/*
 
+RUN echo "alias lla='ls -lAhG --color=auto'" >> ~/.bashrc
 WORKDIR /app
 
-# Install Ruby deps
-ENV JEKYLL_ENV=development
-RUN gem install bundler
-COPY Gemfile Gemfile.lock ./
-RUN bundle config set force_ruby_platform true
-RUN bundle install
 
-# Install Node deps
-ENV NODE_ENV=development
-COPY package.json package-lock.json ./
-RUN npm ci
 
-COPY ./ ./
+# ============== INSTALL FLUTTER ==============
+# TODO recommended to move to a SHA based install for security reasons
+FROM base AS flutter
 
-# So that we have the most up to date submodules
-RUN git submodule update --init --recursive
+# TODO to avoid issues/clashes with the host repo, and to one day remove 
+# the flutter submodule in the host repo, test with clone. Only issue we need 
+# more information on is why the local version would ever be detached or pinned 
```

Comment:
This shouldn't happen anymore since we introduced Dependabot to keep all of the submodules up to date. So for most of the time, the flutter submodule be up to date with the latest on flutter/flutter, with the exception of maybe a ~24 hour delay if we haven't gottena round to reviewing the Dependabot PRs.

Reply:
This was formerly to ensure that if the repo was pulled fresh and the submodules were NOT initialized, they would be correctly set up (should have been included in the build stage this round, so incomplete anyway!). This line though is likely why the current `build.yml` is not broken right now for builds with the `submodules: recursive` missing in the `deploy` job. I agree though, removing (next commit) in favor of it be performed by build file and manually via host setup for local dev.

Reply:
It's in the README, so I feel like that is probably enough?

Reply:
Added comment in this [PR](https://github.com/flutter/website/pull/7737)